### PR TITLE
'Solution'

### PR DESCRIPTION
**Pull request by dsokolovskiy**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1 +1,39 @@
...
+
+    def __init__(self, name: str) -> None:
+        self.name: str = name
+        self.health: int = 100
+        self._hidden: bool = False
+        Animal.alive.append(self)
+
+    @property
+    def hidden(self) -> bool:
+        return self._hidden
```
Comment (Oleg-s-s) [inline app/main.py:15]: What is the purpose of creating `_hidden` and then accessing through `hidden`?)
I dont see in the task that we need private hidden attr

**Review by Oleg-s-s — CHANGES_REQUESTED** [review]

File: tests/test_main.py [hunk 2]
```diff
@@ -144,13 +144,17 @@ def test_print_animal_alive():
...
+    animal = Herbivore("Rex")
+    assert animal.name == "Rex"
+    assert animal.health == 100
+    assert animal.hidden is False
+    assert len(Animal.alive) == 1
+
+    for _ in range(3):
+        Carnivore.bite(animal)
+
+    assert len(Animal.alive) == 0
```
Comment (DevSpaciX) [inline tests/test_main.py:157]: `Never` change the test code, please return everything as it was

File: app/main.py [hunk 3]
```diff
@@ -1 +1,35 @@
...
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+    @staticmethod
+    def bite(herbivore: "Herbivore") -> None:
```
Comment (DevSpaciX) [inline app/main.py:31]: Why do you using " " here ?)

File: app/main.py [hunk 4]
```diff
@@ -1 +1,35 @@
...
+        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
+            herbivore.health -= 50
+            if herbivore.health <= 0:
+                herbivore.die()
```
Comment (DevSpaciX) [inline app/main.py:35]: You just can call the remove method here and it will work I guess

**Review by DevSpaciX — CHANGES_REQUESTED** [review]

**Pull request closed without merging**